doc/guides/maintaining-npm.md

```diff
@@ -0,0 +1,66 @@
+# Maintaining npm in Node.js
+
+## Step 1: Clone npm
+
+```console
+$ git clone https://github.com/npm/npm.git
+$ cd npm
+```
+
+or if you already have npm cloned make sure the repo is up to date
+
+```console
+$ git remote update -p
+$ git reset --hard origin master
+```
+
+## Step 2: Build release
+
+```console
+$ git checkout vX.Y.Z
+$ make release
+```
+
+## Step 3: Remove old npm
+
+```console    
+$ cd /path/to/node
+$ git checkout master
+$ git remote update -p
+$ git reset --hard origin master
+$ git checkout -b npm-x.y.z
+$ cd deps
+$ rm -rf npm
+```
+
+## Step 4: Extract and commit new npm
+
+```console
+$ tar zxf /path/to/npm/release/npm-x.y.z.tgz
+$ git add -f npm
+$ git commit -m "deps: upgrade npm to x.y.z"
+$ cd ..
+```
+
+## Step 5: Update licenses
+
+```console
+$ ./configure
+$ make -j4
+$ ./tools/license-builder.sh
+# The following commands are only neccessary if there are changes
+$ git add .
+$ git commit -m "doc: update npm LICENSE using license-builder.sh"
```

Comment:
It might be worth warning folks that this might end up including changes to -other- licenses, and that wouldn't be appropriate for this commit. They should make sure to check that any changes are actually npm-check related.